connector/src/main/java/org/geysermc/connector/network/translators/java/JavaDeclareCommandsTranslator.java

```diff
@@ -189,38 +232,52 @@ private CommandParamType mapCommandType(CommandParser parser) {
                 return CommandParamType.JSON;
 
             case RESOURCE_LOCATION:
+            case FUNCTION:
                 return CommandParamType.FILE_PATH;
 
-            case INT_RANGE:
-                return CommandParamType.INT_RANGE;
+            //case INT_RANGE:
+            //    return CommandParamType.INT_RANGE; - no longer registered
 
             case BOOL:
-            case DOUBLE:
+                return ENUM_BOOLEAN;
+
+            case OPERATION: // ">=", "==", etc
+                return CommandParamType.OPERATOR;
+
+            case BLOCK_STATE:
+                return BlockTranslator.getAllBlockIdentifiers();
+
+            case ITEM_STACK:
+                return ItemRegistry.ITEM_NAMES;
+
+            case ITEM_ENCHANTMENT:
+                return Enchantment.ALL_JAVA_IDENTIFIERS; //TODO: inventory branch use Java enums
+
+            case ENTITY_SUMMON:
+                return EntityType.ALL_JAVA_IDENTIFIERS;
+
+            case COLOR:
+                return VALID_COLORS;
+
+            case SCOREBOARD_SLOT:
+                return VALID_SCOREBOARD_SLOTS;
+
             case STRING:
             case VEC2:
-            case BLOCK_STATE:
             case BLOCK_PREDICATE:
-            case ITEM_STACK:
             case ITEM_PREDICATE:
-            case COLOR:
             case COMPONENT:
             case OBJECTIVE:
             case OBJECTIVE_CRITERIA:
-            case OPERATION: // Possibly OPERATOR
             case PARTICLE:
-            case ROTATION:
-            case SCOREBOARD_SLOT:
             case SCORE_HOLDER:
             case SWIZZLE:
             case TEAM:
             case ITEM_SLOT:
             case MOB_EFFECT:
-            case FUNCTION:
             case ENTITY_ANCHOR:
             case RANGE:
             case FLOAT_RANGE:
-            case ITEM_ENCHANTMENT:
-            case ENTITY_SUMMON:
             case DIMENSION:
             case TIME:
```

Comment:
We can probably just get rid of all these since they are caught by `default` anyway.

Reply:
Great idea.